core/src/main/scala/cats/Semigroupal.scala

```diff
@@ -70,8 +70,19 @@ object Semigroupal extends ScalaVersionSpecificSemigroupalInstances with Semigro
   implicit def catsSemigroupalForId: Semigroupal[Id] = catsInstancesForId
   implicit def catsSemigroupalForOption: Semigroupal[Option] = cats.instances.option.catsStdInstancesForOption
   implicit def catsSemigroupalForTry: Semigroupal[Try] = cats.instances.try_.catsStdInstancesForTry
+
+  /**
+   * @deprecated
+   *   Any non-pure use of [[scala.concurrent.Future Future]] with Cats is error prone
+   *   (particularly the semantics of [[cats.Traverse#traverse traverse]] with regard to execution order are unspecified).
+   *   We recommend using [[https://typelevel.org/cats-effect/ Cats Effect `IO`]] as a replacement for ''every'' use case of [[scala.concurrent.Future Future]].
```

Comment:
I endorse the advertisement of CE, but Monix is good too, maybe it's also worth mentioning?

Reply:
Great point, sure! @danicheg what is the specific, referentially-transparent datatype we should recommend here, would it be `Task`? Is there any concern to recommend Monix before it has adopted CE3 since it cannot interop with maintained versions of the major libraries?

Reply:
To be fair, anyone who would have used `Future` would also have run into some interop problems 😂

Reply:
I think we could advise using some abstract `IO` datatype and refer to CE and Monix (to libraries, not their concrete datatypes). WDYT?

Reply:
I think the kind of user who might be unintentionally using `Future` with Cats doesn't necessarily have the experience to appreciate what an "abstract `IO` datatype" is.

For the record, I have essentially no experience with Monix, but it seems to me there are an overwhelming number of datatypes in there, not all of which are RT. So I think it would be helpful to be more a little more specific.

Reply:
I think a clear and direct recommendation for `IO` is best here. A user who is using `Future` in a way that's not aware of the RT implications isn't ready for writing tagless style code, so that's out. Monix isn't updated to CE3 and might not be a long time, so that's out (at least for now).

So, `IO`